[Fixes #266] Persisting of GeoNode and Celery logs

jkariscodes opened the pull request:
Edited docker config to allow persistence of geonode [log](https://github.com/GeoNode/geonode-project/blob/7393bb71b14691d4ce135d84fd3c1b1b0c9c2eb5/src/uwsgi.ini#L4) and the celery [log] to a persistent volume called `logs`.

Commit:
GeoNode and Celery logs environment variables

Added GeoNode and Celery logs to output to `stderr` throuth their environment vatiables.

Commit:
Added new volume for logs

Added a new docker volume configuration to save GeoNode logs for monitoring.

Commit:
Update entrypoint.sh

Configured celery logs to output to `stderr`

Commit:
Merge branch 'GeoNode:master' into persist_geonode_logs

randomorder reviewed (approved):
Looks good to me, thanks

frafra commented:
```
GEONODE_LOG_STDOUT=true
CELERY_LOG_STDOUT=true
```

These two variables are defined, but not used.
Wouldn't it better to just have a local switch for logging to STDOUT/STDERR or to the usual file log path?

giohappy commented on src/entrypoint.sh:
```diff
@@ -9,7 +9,11 @@ invoke () {
         /usr/local/bin/invoke $@
     else
-        /usr/local/bin/invoke $@ >> /usr/src/{{project_name}}/invoke.log 2>&1
+        if [ $IS_CELERY = 'true'] || [ $IS_CELERY = 'True' ]
```
What's the purpose of this new block? 
Each container already has its own local invoke log file `/usr/src/{{project_name}}/invoke.log`, why are you differentiating the name in `/usr/src/geonode_proj/invoke.log` for GeoNode and `/usr/src/{{project_name}}/invoke.log` for Celery?

jkariscodes replied:
I deemed them as separate log files and checking whether the log output comes from a celery container or not distunguished by `IS_CELERY` env variable.

giohappy replied:
these logs are saved locally inside the container (`/usr/src/{{project_name}}/invoke.log`), so we already know which container they refer to. These are distinct from the actual logs, that are sent to `/var/log`.

giohappy commented on .env.sample:
```diff
@@ -9,6 +9,8 @@ BACKUPS_VOLUME_DRIVER=local
 FORCE_REINIT=false
 INVOKE_LOG_STDOUT=true
+GEONODE_LOG_STDOUT=true
+CELERY_LOG_STDOUT=true
```
I don't see these new `GEONODE_LOG_STDOUT` and `CELERY_LOG_STDOUT` ever used.
Their purpose should have been to either send logs to a file or redirect to stdout.
For GeoNode this would been tweaking the [uwsgi logger](https://github.com/GeoNode/geonode-project/blob/7393bb71b14691d4ce135d84fd3c1b1b0c9c2eb5/src/uwsgi.ini#L4) 
For Celery it means working on [this line](https://github.com/GeoNode/geonode-project/blob/master/src/celery.sh#L2).

If it's to complicated we can leave this option out for the moment, and just implement the persistent logging into a volume.

jkariscodes replied:
I propose to leave these `GEONODE_LOG_STDOUT` and `CELERY_LOG_STDOUT` options out for the moment.

giohappy commented on docker-compose.yml:
```diff
@@ -14,6 +14,7 @@ x-common-django:
     - backup-restore:/backup_restore
     - data:/data
     - tmp:/tmp
+    - logs:/var/log/
```
are we sure that having a shared `/var/log` root folder between multiple containers is fine? This is a standard path for many services and applications, beyond GeoNode and Celery. I fear that might race to write to the same positions. Is my suspect wrong?

Look how many people write inside /var/log, for example  inside the GeoNode container:
![image](https://user-images.githubusercontent.com/571129/167817504-9e1bcb49-04ca-4481-83b4-c002f992d615.png)

giohappy replied:
I don't know why this comment was marked as resolved but, to me, it isn't resolved. Have you tested the effects of having a shared `/var/log` between multiple containers @jkariukidev?

jkariscodes replied:
Yes. In this case, different services are writing to the directory at the same time and would be better to distinguish the services present in GeoNode where they write to. I have tested with `/opt` which can be accessed by various services of geonode.
![image](https://user-images.githubusercontent.com/23359514/168577125-5eaa8f6f-aac7-49ef-a8b0-719f470ff3d8.png)
and the logs are being written in the following fashion. 
![image](https://user-images.githubusercontent.com/23359514/168576930-47bc06b7-22f5-453b-9a79-28e6ce3d22a3.png)

Commit:
Removed GeoNode/Celery log variables

Removed the GeoNode and Celelery log environment variables to redirect to standard output.

Commit:
Restored the invoke log output

afabiani commented on src/entrypoint.sh:
```diff
@@ -9,7 +9,7 @@ invoke () {
         /usr/local/bin/invoke $@
     else
-        /usr/local/bin/invoke $@ >> /usr/src/{{project_name}}/invoke.log 2>&1
+        /usr/local/bin/invoke $@ >> /usr/src/geonode_proj/celery.log 2>&1
```
`geonode_proj` should be reverted to `{{project_name}}`

jkariscodes replied:
Done

giohappy replied:
@jkariukidev `celery.log` should be reverted to `invoke.log`. This log is not specific to celery.
No need to touch parts that are not related to the tasks of this issue....

Commit:
Update entrypoint.sh

jkariscodes reviewed (commented):
I have made the changes.

Commit:
Update entrypoint.sh
